src/NServiceBus.Core/Routing/DirectRoutingTable.cs

```diff
@@ -0,0 +1,68 @@
+namespace NServiceBus.Routing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Manages the direct routing table.
+    /// </summary>
+    public class DirectRoutingTable
+    {
+        List<Func<Type, IEnumerable<DirectRoutingDestination>>> rules = new List<Func<Type, IEnumerable<DirectRoutingDestination>>>();
```

Comment:
I like that but not sure if this should go in this pull. It is already too big :(